Fix recursion error when getting error descriptions

marcoskirsch opened the pull request:
[X] This contribution adheres to [CONTRIBUTING.md]

### What does this Pull Request accomplish?

* Fix #205 
* Adds function add-on metadata that indicates that a function is_error_handling
   * helper.py function fills in the information (default is False) if missing.
* Marks a few functions as codegen_method : 'No'
   * ClearError()
   * error_message()
   * There are surely more we want to stop generating. That is outside the scope of this PR and will be handled once we work on code coverage for Library in the system tests.
* Adds is_error_handling parameter to errors.handle_error
   * This way it knows not to try and get error description again.
* Re-enable and fix unit test that caught this problem in the first place.

### Why should this Pull Request be merged?

Bugfixes!

### What testing has been done?

Unit tests were run locally (tox)

texasaggie97 commented:
```
============================= test session starts =============================
platform win32 -- Python 3.6.2, pytest-3.2.2, py-1.4.34, pluggy-0.4.0 -- c:\jenkins\workspace\nimi_python_system_tests\.tox\system_tests\scripts\python.exe
cachedir: .cache
rootdir: C:\Jenkins\workspace\nimi_python_system_tests, inifile: tox.ini
collecting ... collected 21 items

bin/niswitch/system_tests/test_system_niswitch.py::test_relayclose PASSED
bin/niswitch/system_tests/test_system_niswitch.py::test_channel_connection PASSED
bin/niswitch/system_tests/test_system_niswitch.py::test_vi_boolean_attribute PASSED
bin/niswitch/system_tests/test_system_niswitch.py::test_vi_string_attribute PASSED
bin/niswitch/system_tests/test_system_niswitch.py::test_vi_int32_attribute PASSED
bin/niswitch/system_tests/test_system_niswitch.py::test_vi_real64_attribute PASSED
bin/niswitch/system_tests/test_system_niswitch.py::test_write_only_attribute PASSED
bin/niswitch/system_tests/test_system_niswitch.py::test_method_reset PASSED
bin/niswitch/system_tests/test_system_niswitch.py::test_method_set_path PASSED
bin/niswitch/system_tests/test_system_niswitch.py::test_method_can_connect PASSED
bin/niswitch/system_tests/test_system_niswitch.py::test_method_reset_with_defaults PASSED
bin/niswitch/system_tests/test_system_niswitch.py::test_functions_get_relay_name PASSED
bin/niswitch/system_tests/test_system_niswitch.py::test_functions_get_channel_name PASSED
bin/niswitch/system_tests/test_system_niswitch.py::test_functions_revision_query PASSED
bin/niswitch/system_tests/test_system_niswitch.py::test_functions_get_next_coercion_record PASSED
bin/niswitch/system_tests/test_system_niswitch.py::test_functions_get_next_interchange_warning PASSED
bin/niswitch/system_tests/test_system_niswitch.py::test_functions_self_test PASSED
bin/niswitch/system_tests/test_system_niswitch.py::test_functions_get_path PASSED
bin/niswitch/system_tests/test_system_niswitch.py::test_functions_error_query PASSED
bin/niswitch/system_tests/test_system_niswitch.py::test_functions_error_message FAILED
bin/niswitch/system_tests/test_system_niswitch.py::test_functions_get_error_description PASSED

================================== FAILURES ===================================
________________________ test_functions_error_message _________________________

session = <niswitch.session.Session object at 0x000002942EB313C8>

    def test_functions_error_message(session):
>       message = session.error_message(-1074126847)
E       AttributeError: 'Session' object has no attribute 'error_message'

bin\niswitch\system_tests\test_system_niswitch.py:144: AttributeError
===================== 1 failed, 20 passed in 0.92 seconds =====================
```

marcoskirsch commented:
Interesting, and nice catch by system tests. Will need to figure out why this is not caught by NI-FAKE unit tests.

marcoskirsch commented:
Removed the test that was failing. It was calling the method that I removed. I still think removing is the right thing to do. @Fladolcetta if you disagree let me know.

texasaggie97 commented on build/helper.py:
```diff
@@ -126,6 +126,17 @@ def _add_python_return_type(f):
 
+def _add_is_error_handling(f):
+    '''Adds is_error_handling information to the function metadata if it isn't already defined. Defaults to False.'''
+    # TODO(marcoskirsch): The information is added in functions_addon.py. I think we can instead infer from method
```
My two <pick your currency> is this way (being explicit) is less error prone.
